feat(MA): enhance data source configurations with status indicators

jabahamondes opened the pull request:
## Changes

- Updated IntegrationFilter, NativeExternalDataSourceConfiguration, NonNativeExternalDataSourceConfiguration, SelfManagedExternalDataSourceConfiguration, and SharedExternalDataSourceConfiguration to utilize sources with status.
- Introduced StatusIcon component to visually represent the status of data sources.
- Added logic to fetch and display the status of sources, improving user feedback on integration states.


## How did you test this code?
Manually
<img width="757" height="611" alt="Screenshot 2025-10-16 at 7 38 07 PM" src="https://github.com/user-attachments/assets/1ef34c39-0290-4b01-bef8-d95f7f90ae99" />
<img width="991" height="376" alt="Screenshot 2025-10-16 at 7 38 44 PM" src="https://github.com/user-attachments/assets/c183f3d9-c8d9-4c2a-b85c-6e5b578fdca6" />
<img width="359" height="268" alt="Screenshot 2025-10-16 at 7 34 19 PM" src="https://github.com/user-attachments/assets/a88deb04-eb25-48d0-912d-dcebf4cf538e" />

Commit:
feat(marketing): add MarketingAnalyticsAggregatedQuery and related components

- Introduced  to the schema and updated relevant types.
- Added  component to handle aggregated marketing analytics data.
- Implemented utility functions for identifying aggregated queries.
- Updated  to utilize the new  for displaying marketing analytics.
- Enhanced query handling in  to support the new aggregated query type.
- Created  for executing the new query type.
- Refactored existing components to improve code organization and reduce duplication.

This update enhances the marketing analytics capabilities by allowing for aggregated data queries, improving the overall analytics experience.

Commit:
fixing mypy baseline errors

github-actions[bot] committed:
Update E2E screenshots for Playwright

github-actions[bot] committed:
Update E2E screenshots for Playwright

Commit:
Pr's comments

Commit:
Merge branch 'feat/adding-aggregated-marketing-overview' of github.com:PostHog/posthog into feat/adding-aggregated-marketing-overview

Commit:
refactor(marketing): clean up marketing analytics logic and improve query handling

- Removed unnecessary comments in marketing analytics tiles logic.
- Refactored methods in MarketingAnalyticsAggregatedQueryRunner to simplify column handling and improve clarity.
- Updated method names for better understanding of their functionality.
- Enhanced the exclusion logic for specific metrics in column mappings.

These changes streamline the codebase and enhance maintainability while ensuring the functionality remains intact.

Commit:
feat(marketing): add Marketing Analytics Aggregated query type to insights

- Introduced a new query type for Marketing Analytics Aggregated, enhancing the insights capabilities.
- Updated the QUERY_TYPES_METADATA to include the new query type with its name, icon, and menu visibility.

This addition improves the flexibility of analytics queries available in the application.

Commit:
feat(marketing): add IntegrationFilter for enhanced marketing analytics

- Introduced IntegrationFilter to allow filtering by selected integration source IDs in marketing analytics queries.
- Updated schema and TypeScript interfaces to include integrationFilter in relevant query types.
- Implemented IntegrationFilter component for user interaction in the UI, enabling selection of integration sources.
- Enhanced marketing analytics logic to support integration filtering in data queries.

This addition improves the flexibility and usability of marketing analytics by allowing users to filter data based on specific integrations.

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
Merge conflicts

Commit:
Merge conflicts

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-integration-filters-into-ma-dashboard

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-integration-cta-in-dropdown

Commit:
feat(marketing-analytics): add AddIntegrationButton component and integrate into MarketingAnalyticsFilters

- Introduced  for adding new data sources with categorized integration options.
- Updated  to include  for better visual representation.
- Integrated  into  for improved user experience.

jabahamondes committed:
Update frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/MarketingAnalyticsFilters/AddIntegrationButton.tsx

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

jabahamondes committed:
Update frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/MarketingAnalyticsFilters/AddIntegrationButton.tsx

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
feat(marketing-analytics): enhance data source configurations with status indicators

- Updated IntegrationFilter, NativeExternalDataSourceConfiguration, NonNativeExternalDataSourceConfiguration, SelfManagedExternalDataSourceConfiguration, and SharedExternalDataSourceConfiguration to utilize sources with status.
- Introduced StatusIcon component to visually represent the status of data sources.
- Added logic to fetch and display the status of sources, improving user feedback on integration states.

Commit:
Merge conflicts

Commit:
Merge branch 'feat/adding-integration-cta-in-dropdown' into feat/adding-statuses-in-ma-settings-and-dashboard-integration

Commit:
Merge conflicts

Commit:
removing unnecessary comment

Commit:
fix: improve status handling in IntegrationFilter and StatusIcon components

- Updated IntegrationFilter to only show the status icon if the source has a status and a status message, excluding 'Completed' sources.
- Enhanced StatusIcon to throw an error for unknown statuses instead of returning an empty object.
- Refined marketingAnalyticsLogic to ensure 'Completed' status is returned only when all required fields have synced.

lricoy commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/logic/marketingAnalyticsLogic.ts:
```diff
@@ -37,6 +37,56 @@ import {
+    source: { id: string; name: string; type: string; prefix?: string },
+    nativeSources: ExternalDataSource[],
+    validExternalTables: ExternalTable[]
+): { status: 'Completed' | 'Failed' | 'Running' | 'warning' | 'error'; message: string } {
```
We could use a `StatusSource` type here, but really nit, I just also tend to create those and the autocomplete works but when we're reviewing the magic strings look weird. It is the same with insight props, but I think we should default to use more standard data structure/types

lricoy reviewed (approved):
LGTM, just some nit comments that popped in my head.

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-statuses-in-ma-settings-and-dashboard-integration

Commit:
feat: enhance status handling in marketing analytics components

- Updated IntegrationFilter to include new status checks for ExternalDataSchemaStatus and MarketingSourceStatus.
- Refactored SharedExternalDataSourceConfiguration to utilize the new status types and improve type safety.
- Modified StatusIcon to accept the new SourceStatus type, ensuring consistent status representation across components.
- Adjusted logic in marketingAnalyticsLogic to return appropriate statuses based on the new enum definitions.

Commit:
Fix type checking
